packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.test.js

```diff
@@ -162,5 +162,47 @@ describe('<TrapFocus />', () => {
       clock.tick(500); // wait for the interval check to kick in.
       expect(dialog).toHaveFocus();
     });
+
+    describe('prop: disableAutoFocus', () => {
+      it('should not focus the trap focus root', () => {
+        render(
+          <TrapFocus {...defaultProps} open disableAutoFocus>
+            <div tabIndex={-1} data-testid="modal" />
+          </TrapFocus>,
+        );
+
+        clock.tick(500); // wait for the interval check to kick in.
+        expect(initialFocus).toHaveFocus();
+      });
+
+      it('should only trap focus once the focus moves inside', () => {
+        const Test = (props) => (
+          <div>
+            <input />
+            <TrapFocus {...defaultProps} open disableAutoFocus {...props}>
+              <div tabIndex={-1} data-testid="modal" />
+            </TrapFocus>
+          </div>
+        );
+
+        const { getByRole, getByTestId, setProps } = render(<Test />);
+
+        // the trap doesn't force anyone
+        getByRole('textbox').focus();
+        expect(getByRole('textbox')).toHaveFocus();
```

Comment:
Repeats the previous test incomplete. The previous test is sufficient.

Reply:
It doesn't repeat the previous test. You can verify this affirmation by running the new tests without the changes from 44290b409c557e71febcae7dfdc670a53e97b590. It will fail at this line. The reason is that in test case 1. no focus event is fired, in this test case 2., a focus event is fired.

Reply:
However, I think that we could improve the organization of it.

Reply:
For instance, it feels like the **description** of the test makes it a duplicate. What about I move this case to test number 1?

Reply:
I'm fairly certain there are 3 tests here not two. At least from the description and code I can't follow what is testing was. This and the previous test both test if an element outside the focus trap can be focused in my eyes.

Reply:
Ok, I will work on a better organization and separation

Reply:
What makes this different from https://github.com/mui-org/material-ui/pull/21612/files#diff-5c5a9f3c6d5e166de202b4f7beb11760R180-R181?

It seems to me both render the same thing.

Reply:
Are you suggesting?

```suggestion
```

The incentive for the asserting is to make sure the focus restores to where it was originally. I'm sorry, your link isn't working, I can't see which line you are referring to.

Reply:
This and the previous test render the same thing and make the same assertion. It's unclear what is different. The link doesn't scroll automatically but it does highlight the lines I'm referring to.

Reply:
The motivation for having an input, in this third test, is to make sure the focus isn't restored to the active element at the moment the trap mounts, but when it activates